trace/trace_test.go

```diff
@@ -83,35 +83,55 @@ func TestTotalTime(t *testing.T) {
 }
 
 func TestLog(t *testing.T) {
-	test := struct {
+	tests := []struct {
 		name             string
+		msg              string
+		msgArgs          []interface{}
 		expectedMessages []string
 		sampleTrace      *Trace
 	}{
-		name: "Check the log dump with 3 msg",
-		expectedMessages: []string{
-			"msg1", "msg2", "msg3",
+		{
+			name: "Check the log dump with 3 msg",
+			expectedMessages: []string{
+				"msg1", "msg2", "msg3",
+			},
+			sampleTrace: &Trace{
+				name: "Sample Trace",
+				steps: []traceStep{
+					{stepTime: time.Now(), msg: "msg1"},
+					{stepTime: time.Now(), msg: "msg2"},
+					{stepTime: time.Now(), msg: "msg3"},
+				},
+			},
 		},
-		sampleTrace: &Trace{
-			name: "Sample Trace",
-			steps: []traceStep{
-				{time.Now(), "msg1"},
-				{time.Now(), "msg2"},
-				{time.Now(), "msg3"},
+		{
+			name: "Check formatting",
+			expectedMessages: []string{
+				"URL=/api,count=3", "msg1 arg1 2 false",
+			},
+			sampleTrace: &Trace{
+				name:        "Sample Trace",
+				details:     "URL=%s,count=%d",
+				detailsArgs: []interface{}{"/api", 3},
+				steps: []traceStep{
+					{stepTime: time.Now(), msg: "msg1 %s %d %t", msgArgs: []interface{}{"arg1", 2, false}},
```

Comment:
Probably worth adding a testcase with multiple trace steps to test new line strings in the buffer?